build/build.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+set -e
+
+SBT_CMD="sbt ++${TRAVIS_SCALA_VERSION} validate"
+
+if [[ "${TRAVIS_PULL_REQUEST}" == "false" ]]; then
+    SBT_CMD+=" +coverageOff +publish"
+    openssl aes-256-cbc -pass env:ENCRYPTION_PASSWORD -in ./build/secring.gpg.enc -out local.secring.gpg -d
+    openssl aes-256-cbc -pass env:ENCRYPTION_PASSWORD -in ./build/pubring.gpg.enc -out local.pubring.gpg -d
+    openssl aes-256-cbc -pass env:ENCRYPTION_PASSWORD -in ./build/credentials.sbt.enc -out local.credentials.sbt -d
+    openssl aes-256-cbc -pass env:ENCRYPTION_PASSWORD -in ./build/deploy_key.pem.enc -out local.deploy_key.pem -d
+
+    if [[ "${TRAVIS_BRANCH}" == "master" && $(cat version.sbt) != *"SNAPSHOT"* ]]; then
+        eval "$(ssh-agent -s)"
+        chmod 600 local.deploy_key.pem
+        ssh-add local.deploy_key.pem
+        git config --global user.name "${CI_DEPLOY_USERNAME}"
+        git config --global user.email "${CI_DEPLOY_EMAIL}"
```

Comment:
What are those `CI_DEPLOY` values by default? Should we set them into something?

Reply:
Perhaps I should somehow use the same email/user name while generating a github deploy key: https://help.github.com/articles/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent/#generating-a-new-ssh-key

Reply:
As we don't really have a Finch domain yet, I made an email alias from my personal domain: `ci@kostyukov.net`. Let's just use it for now as a hard-coded value.

Reply:
Okay cool. How about `Finch CI` for the username so it's easy to distinguish automated CI deploys vs deploys made by you?

Reply:
borrowing ideas from https://github.com/traneio/arrows/blob/master/build/build.sh#L21

Reply:
Yep. `Finch CI` sounds good!